### PR TITLE
Reorg the injection-reconciler

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

potential fix for [K8S-10730](https://dt-rnd.atlassian.net/browse/K8S-10730)

The injection-reconciler is bit of a tangled mess. The two injections types (oneagent and enrichment) are not really independent as they should be, causing unexpected behaviour (mainly ugly logs).

## How can this be tested?
Try that no matter if only OneAgent injection or Metadata-enrichment injection is needed, the operator does not produce error logs and each injection is setup and cleaned up properly

### CODE REVIEW

File: pkg/controllers/dynakube/injection/reconciler.go [hunk 1]
```diff
@@ -131,32 +101,54 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
...
 	if err != nil {
-		return errors.WithMessagef(err, "failed to list namespaces for dynakube %s", r.dk.Name)
+		log.Error(err, "failed to list namespaces for dynakube", "dkName", r.dk.Name)
 	}
 
 	dkMapper := r.createDynakubeMapper(ctx)
-
 	if err := dkMapper.UnmapFromDynaKube(namespaces); err != nil {
-		log.Info("could not unmap DynaKube from namespace")
+		log.Error(err, "could not unmap DynaKube from namespace", "dkName", r.dk.Name)
```
Comment (waodim) [inline pkg/controllers/dynakube/injection/reconciler.go:117]: very minor nitpick (and probably annoying) but here we write "DynaKube" but in the log line above we write "dynakube", maybe we should be consistent there

File: pkg/controllers/dynakube/injection/reconciler.go [hunk 2]
```diff
@@ -174,29 +166,41 @@ func (r *reconciler) setupOneAgentInjection(ctx context.Context) error {
...
-
+func (r *reconciler) cleanupOneAgentInjection(ctx context.Context) {
 	if meta.FindStatusCondition(*r.dk.Conditions(), codeModulesInjectionConditionType) != nil {
-		err := initgeneration.NewInitGenerator(r.client, r.apiReader, r.dk.Namespace).Cleanup(ctx, namespaces)
+		defer meta.RemoveStatusCondition(r.dk.Conditions(), codeModulesInjectionConditionType)
+
+		namespaces, err := mapper.GetNamespacesForDynakube(ctx, r.apiReader, r.dk.Name)
 		if err != nil {
-			errs = append(errs, err)
+			log.Error(err, "failed to clean-up code module injection")
```
[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/controllers/dynakube/injection/reconciler.go]: done

File: pkg/controllers/dynakube/injection/reconciler.go [hunk 3]
```diff
@@ -210,14 +214,21 @@ func (r *reconciler) setupEnrichmentInjection(ctx context.Context) error {
...
-func (r *reconciler) cleanupEnrichmentInjection(ctx context.Context, namespaces []corev1.Namespace) {
+func (r *reconciler) cleanupEnrichmentInjection(ctx context.Context) {
 	if meta.FindStatusCondition(*r.dk.Conditions(), metaDataEnrichmentConditionType) != nil {
-		err := ingestendpoint.NewSecretGenerator(r.client, r.apiReader, r.dk.Namespace).RemoveEndpointSecrets(ctx, namespaces)
+		defer meta.RemoveStatusCondition(r.dk.Conditions(), metaDataEnrichmentConditionType)
+
+		namespaces, err := mapper.GetNamespacesForDynakube(ctx, r.apiReader, r.dk.Name)
 		if err != nil {
 			log.Error(err, "failed to clean-up metadata-enrichment secrets")
+
```
[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/controllers/dynakube/injection/reconciler.go:224]: done

**Review by waodim — APPROVED** [review]
Deployed operator as wanted, got no errors, logs look ok, clean up looks ok

**Review by aorcholski — APPROVED** [review]
LGTM, I see no error messages

**Pull request merged**